src/types.ts

```diff
@@ -528,73 +528,99 @@ export interface AdminStats {
   /**
    * Total balance of all outputs in sats in the database
    */
-  balance: number
+  balance: number;
   /**
    * Number of destinations in the database
    */
-  destinations: number
+  destinations: number;
   /**
    * Number of transactions in the database
    */
-  transactions: number
+  transactions: number;
   /**
    * Number of paymail addresses in the database
    */
-  paymail_addresses: number
+  paymail_addresses: number;
   /**
    * Number of utxos in the database
    */
-  utxos: number
+  utxos: number;
   /**
    * Number of xpubs registered in the database
    */
-  xpubs: number
+  xpubs: number;
   /**
    * A key value object of dates and number of transactions on that date (YYYYMMDD)
    */
-  transactions_per_day: { [key: string]: any }
+  transactions_per_day: { [key: string]: any };
   /**
    * Number of utxos per output type
    */
-  utxos_per_type: { [key: string]: any }
+  utxos_per_type: { [key: string]: any };
 }
 
 /**
  * Key basic information
  */
 export interface Key {
-  xPriv(): string
-  xPub: PubKey
+  xPriv(): string;
+  xPub: PubKey;
 }
 
 /**
  * Extends Key interface with mnemonic information
  */
 export interface KeyWithMnemonic extends Key {
-  mnemonic: string
+  mnemonic: string;
 }
 
 /**
  * Public key information
  */
 export interface PubKey {
-  toString(): string
+  toString(): string;
 }
 
 /**
  * Client options for instantiating a new SPV Wallet client
  */
-export interface ClientOptions {
-  accessKeyString?: string
-  accessKey?: bsv.PrivateKey
-  adminKey?: string
-  debug?: boolean
-  signRequest?: boolean
-  xPriv?: bsv.HDPrivateKey
-  xPrivString?: string
-  xPub?: bsv.HDPublicKey
-  xPubString?: string
-}
+export type ClientOptions =
+  | {
+      /**
+       * Making request with the 'xPub' doesn't sign requests, which means that the xPub will be placed into the request header.
+       */
+      signRequest: false;
+
+      /**
+       * With xPub you can make non-admin requests without signing.
+       * The SendToRecipients function will not work because it requires the xPriv.
+       */
+      xPub: string;
```

Comment:
As in the code, actually if we have xPriv, then we're still able to get xPub and therefore such an options should also be valid

Reply:
New `ClientOptions` has been applied - as we discussed 😄